plugins/inputs/ctrlx_datalayer/ctrlx_datalayer_subscription_test.go

```diff
@@ -0,0 +1,315 @@
+package ctrlx_datalayer
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+func TestSubscription_createRequest(t *testing.T) {
+	type fields struct {
+		Nodes             []Node
+		Tags              map[string]string
+		Measurement       string
+		PublishInterval   config.Duration
+		KeepaliveInterval config.Duration
+		ErrorInterval     config.Duration
+		SamplingInterval  config.Duration
+		QueueSize         uint
+		QueueBehaviour    string
+		DeadBandValue     float64
+		ValueChange       string
+		OutputJSONString  bool
+	}
```

Comment:
No need to redefine the `Subscription` struct, I would just use that one.

Reply:
Done, see https://github.com/influxdata/telegraf/pull/11155/commits/840b36589ab52f80bb125e47108d403c0a97414f

Reply:
The struct is still there?

Reply:
Sorry, we made several commits and removed the struct with https://github.com/influxdata/telegraf/pull/11155/commits/f76c9e9525f1c73efdc17b04d4a29fdefdc5e10c